### PR TITLE
Navigation: Collapsible section nav implementation

### LINKED ISSUE #55724: Navigation: Collapsable section nav

### LINKED ISSUE #55725: Navigation: `SectionNav` for mobile/small screens

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- removes `NavBarToggle` from megamenu
- adds a new component, `SectionNavToggle`. this is basically `NavBarToggle` but allows us to make changes if we want to (with the view of removing `NavBarToggle` once we turn `topnav` on)
- adds `SectionNavToggle` to `PageNew`, allowing it to collapse/expand the `SectionNav` and storing the state in localstorage.
- handles mobile behaviour in an ok way - allows the section to expand to 50% of the view height and has scroll behaviour within

# before
https://user-images.githubusercontent.com/20999846/192997665-fda1c2f9-a104-4be2-bdf0-469922a0cf33.mp4

# after
https://user-images.githubusercontent.com/20999846/192997435-58fc1126-d80f-4977-b17f-12c62267b9a5.mp4

**Which issue(s) this PR fixes**:

Fixes #55724
Fixes #55725

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by torkelo** [comment]
@ashharrison90 can you explore auto collapsing the section nav for smaller screens (below lg)?

**Comment by ashharrison90** [comment]
@torkelo have added some logic to collapse below a certain size. it was a little fiddlier than i imagined tbh 😅 the behaviour is something like:

above the `lg` breakpoint:
- section nav will be expanded/collapsed based on your preference set in `localStorage`, this will default to `true`
- section nav will remain in that state on page transitions

below the `lg` breakpoint:
- section nav will be automatically collapsed. you can manually expand it
- it will collapse again once you moved to another page (e.g. click something in the section nav)

dragging between sizes around the `lg` boundary:
- section nav will always collapse below `lg`
- it **may** expand again above `lg`, depending on your preference set in `localStorage`

lemme know if that's the behaviour you were after 👍

**Comment by torkelo** [comment]
@ashharrison90 that sounds good!

File: public/app/core/components/PageNew/Page.tsx [hunk 1]
```diff
@@ -29,10 +31,20 @@ export const Page: PageType = ({
   scrollRef,
   ...otherProps
 }) => {
+  const theme = useTheme2();
   const styles = useStyles2(getStyles);
   const navModel = usePageNav(navId, oldNavProp);
   const { chrome } = useGrafana();
 
+  const mediaQuery = window.matchMedia(`(max-width: ${theme.breakpoints.values.lg}px)`);
+  mediaQuery.addEventListener('change', (e) => setNavExpanded(e.matches ? false : navExpandedPreference));
```
Comment (torkelo) [inline public/app/core/components/PageNew/Page.tsx:40]: Should be a way to do this normal css media query I think (ie no js code on size change) 

This looks like it adds event listeners on every render?

Comment (ashharrison90) [reply public/app/core/components/PageNew/Page.tsx]: yikes, very good point about the event listeners, fixed that

i'm not sure the same behaviour is possible with just css media queries 🤔 couldn't make it work today, but i'll try again tomorrow 👍

**Review by torkelo — APPROVED** [review]

**Review by lpskdl — APPROVED** [review]

**Pull request merged**